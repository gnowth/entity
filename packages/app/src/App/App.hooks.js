import _ from 'lodash';
import idx from 'idx';
import React from 'react';
import { DefaultProvider } from '@gnowth/default';

export default {
  useComponents(props, context) {
    return {
      DefaultProvider: props.defaults ? DefaultProvider : React.Fragment,
      IntlProvider: (props.intlProviderProps && context.intlProvider) || React.Fragment,
      ThemeProvider: (props.themeProviderProps && context.themeProvider) || React.Fragment,
    };
  },

  usePropsDefault(props, context, provider) {
    const defaults = React.useMemo(
      () => (
        _.isFunction(props.defaults)
          ? props.defaults(context.defaults)
          : { ...context.defaults, ...props.defaults }
      ),
      [context.defaults, props.defaults],
    );

    return provider === React.Fragment
      ? undefined
      : defaults;
  },

  usePropsIntl(props, context, provider) {
    const intlProviderProps = React.useMemo(
      () => (
        _.isFunction(props.intlProviderProps)
          ? props.intlProviderProps(context.intlProviderProps)
          : {
            ...context.intlProviderProps,
            ...props.intlProviderProps,
            defaultFormats: {
              ...idx(context, x => x.intlProviderProps.defaultFormats),
              ...idx(props, x => x.intlProviderProps.defaultFormats),
            },
            formats: {
              ...idx(context, x => x.intlProviderProps.formats),
              ...idx(props, x => x.intlProviderProps.formats),
            },
            messages: {
              ...idx(context, x => x.intlProviderProps.messages),
              ...idx(props, x => x.intlProviderProps.messages),
            },
          }
      ),
      [context.intlProviderProps, props.intlProviderProps],
    );

    return provider === React.Fragment
      ? undefined
      : intlProviderProps;
  },

  usePropsTheme(props, context, provider) {
    const theme = React.useMemo(
      () => (
        _.isFunction(props.themeProviderProps)
          ? props.themeProviderProps(context.themeProviderProps)
          : { ...context.themeProviderProps, ...props.themeProviderProps }
      ),
      [context.themeProviderProps, props.themeProviderProps],
    );

    return provider === React.Fragment
      ? undefined
      : theme;
  },
};
