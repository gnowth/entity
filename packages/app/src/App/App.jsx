import PropTypes from 'prop-types';
import React from 'react';

import { AppProvider, Context } from '../context';
import defaultHooks from './App.hooks';

function App(props) {
  const hooks = { ...defaultHooks, ...props.hooks };
  const context = React.useContext(Context);
  const { DefaultProvider, IntlProvider, ThemeProvider } = hooks.useComponents(props, context);

  const nextContext = {
    ...context,
    defaults: hooks.usePropsDefault(props, context, DefaultProvider),
    intlProviderProps: hooks.usePropsIntl(props, context, IntlProvider),
    themeProviderProps: hooks.usePropsTheme(props, context, ThemeProvider),
  };

  return (
    <AppProvider {...nextContext}>
      <IntlProvider {...nextContext.intlProviderProps}>
        <ThemeProvider {...nextContext.themeProviderProps}>
          <DefaultProvider {...nextContext.defaults}>
            { props.children }
          </DefaultProvider>
        </ThemeProvider>
      </IntlProvider>
    </AppProvider>
  );
}

App.propTypes = {
  children: PropTypes.node.isRequired,
  defaults: PropTypes.shape({}),
  hooks: PropTypes.exact({
    useComponents: PropTypes.func,
    usePropsDefault: PropTypes.func,
    usePropsIntl: PropTypes.func,
    usePropsTheme: PropTypes.func,
  }),
  intlProviderProps: PropTypes.exact({
    locale: PropTypes.string,
    messages: PropTypes.shape({}),
  }),
  themeProviderProps: PropTypes.exact({
    theme: PropTypes.shape({}),
  }),
};

App.defaultProps = {
  defaults: undefined,
  hooks: undefined,
  intlProviderProps: undefined,
  themeProviderProps: undefined,
};

export default React.memo(App);
