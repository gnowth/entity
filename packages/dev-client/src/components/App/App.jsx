import standardTheme from '@gnowth/theme-standard';
import PropTypes from 'prop-types';
import React from 'react';
import { EntityTitle } from '@entity/core';
import { AppRoot } from '@gnowth/app';
import { GlobalStyles } from '@gnowth/theme';
import { Provider } from 'react-redux';
import { IntlProvider } from '@private/react-intl';
import { ThemeProvider } from 'styled-components';

import * as theme from 'styles';

import defaults from '../../defaults';

const App = props => (
  <AppRoot
    defaults={defaults}
    entity={EntityTitle}
    intlProvider={IntlProvider}
    intlProviderProps={{ locale: 'en', messages: {} }}
    themeProvider={ThemeProvider}
    themeProviderProps={{ theme: { ...standardTheme, ...theme } }}
    {...props}
  >
    <Provider store={defaults.store}>
      { props.children }
    </Provider>

    <GlobalStyles />
  </AppRoot>
);

App.propTypes = {
  children: PropTypes.node.isRequired,
};

export default React.memo(App);
