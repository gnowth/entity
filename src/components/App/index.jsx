import PropTypes from 'prop-types';
import React from 'react';
import { AppRoot } from '@gnowth/app';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';

import * as theme from 'styles';
import GlobalStyles from 'styles/global';

import defaults from '../../defaults';

const App = props => (
  <AppRoot
    defaults={defaults}
    intlProvider={IntlProvider}
    intlProviderProps={{ locale: 'en', messages: {} }}
    themeProvider={ThemeProvider}
    themeProviderProps={{ theme }}
    {...props}
  >
    <Provider store={defaults.store}>
      { props.children }
    </Provider>

    <GlobalStyles />
  </AppRoot>
);

App.propTypes = {
  children: PropTypes.node.isRequired,
};

export default App;
