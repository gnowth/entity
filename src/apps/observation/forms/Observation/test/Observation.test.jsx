import renderer from 'react-test-renderer';
import React from 'react';
import { Fields } from '@entity/core';
import { List } from 'immutable';

import App from 'components/App';
import EntityObservation from 'apps/observation/entities/Observation';

import FormObservation from '..';

jest.mock('store');

it('renders correctly', () => {
  const tree = renderer.create(
    <App>
      <FormObservation
        errors={List()}
        field={new Fields.EntityField({ entity: EntityObservation })}
        onChange={() => undefined}
        value={EntityObservation.dataToRecord({})}
      />
    </App>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
