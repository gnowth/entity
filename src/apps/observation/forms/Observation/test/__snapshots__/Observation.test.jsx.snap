// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX cFlEyZ"
  />
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX cFlEyZ"
  />
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX WEtDF"
  >
    <label
      className="TypeSet-sc-2d00f5-0 jJnJJj Labelstyles__Label-sc-1i1vlit-1 eVfaML"
    >
      Observation Date
    </label>
    <div
      className="Datestyles__Wrapper-ety37l-0 sc-gzVnrw kVNhwm"
    >
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="Datestyles__DatePicker-ety37l-1 bVaXHv"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX WEtDF"
  >
    <label
      className="TypeSet-sc-2d00f5-0 jJnJJj Labelstyles__Label-sc-1i1vlit-1 eVfaML"
    >
      Title
    </label>
    <input
      className="Input__WidgetInput-sc-2m50dl-0 ftREri"
      errors={Immutable.List []}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      name="title"
      onChange={[Function]}
      value=""
      variant="main"
    />
  </div>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX WEtDF"
  >
    <label
      className="TypeSet-sc-2d00f5-0 jJnJJj Labelstyles__Label-sc-1i1vlit-1 eVfaML"
    >
      Short Title
    </label>
    <input
      className="Input__WidgetInput-sc-2m50dl-0 ftREri"
      errors={Immutable.List []}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      name="title_short"
      onChange={[Function]}
      value=""
      variant="main"
    />
  </div>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX WEtDF"
  >
    <label
      className="TypeSet-sc-2d00f5-0 jJnJJj Labelstyles__Label-sc-1i1vlit-1 eVfaML"
    >
      Description
    </label>
    <textarea
      className="Textarea__WidgetTextarea-sc-11zi4lv-0 bROpBJ"
      errors={Immutable.List []}
      name="description"
      onChange={[Function]}
      value=""
      variant="main"
    />
  </div>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX WEtDF"
  >
    <label
      className="TypeSet-sc-2d00f5-0 jJnJJj Labelstyles__Label-sc-1i1vlit-1 eVfaML"
    >
      Due Date
    </label>
    <div
      className="Datestyles__Wrapper-ety37l-0 sc-gzVnrw kVNhwm"
    >
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="Datestyles__DatePicker-ety37l-1 bVaXHv"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX WEtDF"
  >
    <label
      className="TypeSet-sc-2d00f5-0 jJnJJj Labelstyles__Label-sc-1i1vlit-1 eVfaML"
    >
      Follow up actions
    </label>
  </div>
  
  
  <button
    className="Buttonstyles__Button-mk3f56-0 ljupdW"
    onClick={[Function]}
    to=""
  >
    Add action
  </button>
  <div
    className="Labelstyles__UILabelRoot-sc-1i1vlit-0 sc-ifAKCX eWMhR"
  >
    <div
      className="Controlsstyles__Controls-b8kzhl-0 fjcuwP"
    >
      
      
      <button
        className="Buttonstyles__Button-mk3f56-0 ljupdW"
        onClick={[Function]}
        to=""
      >
        RESET
      </button>
      <div
        className="Spacer-sc-1kn0srl-0 lfRbbW"
      />
      
      
      <button
        className="Buttonstyles__Button-mk3f56-0 eISILG"
        onClick={[Function]}
        to=""
      >
        SAVE
      </button>
      
      
      <button
        className="Buttonstyles__Button-mk3f56-0 eISILG"
        onClick={[Function]}
        to=""
      >
        SUBMIT
      </button>
    </div>
  </div>
</div>
`;
